File: NewsFeed/app/src/main/java/com/example/newsfeed/NewsDetailActivity.java
package com.example.newsfeed;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.Html;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

public class NewsDetailActivity extends AppCompatActivity {

    ImageButton back;
    ImageView likeBtn;
    ImageView saveBtn;
    ImageView profilePhoto;
    TextView author;
    ImageView postImage;
    TextView postText;
    TextView date;
    TextView likesCnt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_news_detail);

        back = findViewById(R.id.back);
        likeBtn = findViewById(R.id.likeBtn);
        saveBtn = findViewById(R.id.saveBtn);
        profilePhoto = findViewById(R.id.profilePhoto);
        author = findViewById(R.id.author);
        postImage = findViewById(R.id.postImage);
        postText = findViewById(R.id.postText);
        date = findViewById(R.id.date);
        likesCnt = findViewById(R.id.likesCnt);

        final News news = (News)getIntent().getSerializableExtra("news");
        final int index = getIntent().getIntExtra("index", 0);

        profilePhoto.setImageResource(news.getProfilePhoto());
        postImage.setImageResource(news.getPostImage());

        author.setText(news.getAuthor());
        String s = "<b>"+ news.getAuthor() + "</b>" + " " + news.getPostText();
        postText.setText(Html.fromHtml(s));
        date.setText(news.getDate());
        likesCnt.setText(news.getLikesCnt()+" likes");


        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(NewsDetailActivity.this, MainActivity.class);
                intent.putExtra("index", index);
                startActivity(intent);
            }
        });
        likeBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (!news.isLiked()){
                    likeBtn.setImageResource(R.drawable.liked);
                    news.setLiked(true);
                } else {
                    likeBtn.setImageResource(R.drawable.like);
                    news.setLiked(false);
                }
            }
        });
        saveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (!news.isSaved()){
                    saveBtn.setImageResource(R.drawable.saved);
                    news.setSaved(true);
                } else {
                    saveBtn.setImageResource(R.drawable.save);
                    news.setSaved(false);
                }
            }
        });
    }
}

File: Hw4/app/src/main/java/com/example/hw4/NewsDetailActivity.kt
package com.example.hw4

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.text.Html
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class NewsDetailActivity : AppCompatActivity() {

    var back: ImageButton? = findViewById(R.id.back)
    var profilePhoto: ImageView? = findViewById(R.id.profilePhoto)
    var author: TextView? = findViewById(R.id.author)
    var postImage: ImageView? = findViewById(R.id.postImage)
    var postText: TextView? = findViewById(R.id.postText)
    var date: TextView? = findViewById(R.id.date)
    var likesCnt: TextView? = findViewById(R.id.likesCnt)

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_news_detail)

        val news = intent.getSerializableExtra("news") as News
        val index = intent.getIntExtra("index", 0)
        profilePhoto?.setImageResource(news.profilePhoto)
        postImage?.setImageResource(news.postImage)
        author?.text = news.author
        val s = "<b>" + news.author + "</b>" + " " + news.postText
        postText?.text = Html.fromHtml(s)
        date?.text = news.date
        likesCnt?.text = news.likesCnt.toString() + " likes"

        back?.setOnClickListener {
            val intent = Intent(this@NewsDetailActivity, MainActivity::class.java)
            intent.putExtra("index", index)
            startActivity(intent)
        }

    }
}
